posthog/hogql/database/database.py

```diff
@@ -163,52 +182,22 @@ def create_hogql_database(
         database.events.fields["person"] = FieldTraverser(chain=["pdi", "person"])
         database.events.fields["person_id"] = FieldTraverser(chain=["pdi", "person_id"])
 
-    elif modifiers.personsOnEventsMode == PersonsOnEventsMode.v1_mixed:
-        # person.id via a join, person.properties on events
-        database.events.fields["person_id"] = FieldTraverser(chain=["pdi", "person_id"])
-        database.events.fields["person"] = FieldTraverser(chain=["poe"])
-        database.events.fields["poe"].fields["id"] = FieldTraverser(chain=["..", "pdi", "person_id"])
-        database.events.fields["poe"].fields["created_at"] = FieldTraverser(chain=["..", "pdi", "person", "created_at"])
-        database.events.fields["poe"].fields["properties"] = StringJSONDatabaseField(name="person_properties")
-
-    elif modifiers.personsOnEventsMode == PersonsOnEventsMode.v1_enabled:
-        database.events.fields["person"] = FieldTraverser(chain=["poe"])
+    elif modifiers.personsOnEventsMode == PersonsOnEventsMode.person_id_no_override_properties_on_events:
         database.events.fields["person_id"] = StringDatabaseField(name="person_id")
+        _use_person_properties_from_events(database)
 
-    elif modifiers.personsOnEventsMode == PersonsOnEventsMode.v2_enabled:
-        database.events.fields["event_person_id"] = StringDatabaseField(name="person_id")
-        database.events.fields["override"] = LazyJoin(
-            from_field=["event_person_id"],
-            join_table=PersonOverridesTable(),
-            join_function=join_with_person_overrides_table,
-        )
-        database.events.fields["person_id"] = ExpressionField(
-            name="person_id",
-            expr=parse_expr(
-                "ifNull(nullIf(override.override_person_id, '00000000-0000-0000-0000-000000000000'), event_person_id)",
-                start=None,
-            ),
-        )
+    elif modifiers.personsOnEventsMode == PersonsOnEventsMode.person_id_override_properties_on_events:
+        _use_person_id_from_person_overrides(database)
+        _use_person_properties_from_events(database)
         database.events.fields["poe"].fields["id"] = database.events.fields["person_id"]
-        database.events.fields["person"] = FieldTraverser(chain=["poe"])
-
-    elif modifiers.personsOnEventsMode == PersonsOnEventsMode.v3_enabled:
```

Comment:
This branch was temporarily removed: nobody was flagged into it and #21455 made it unreachable outside of tests/debug. I'll bring it back as a follow up (see comment on `_use_person_id_from_person_overrides`.)